userspace/libsinsp/user.cpp

```diff
@@ -266,6 +266,14 @@ scap_groupinfo *sinsp_usergroup_manager::groupinfo_map_insert(
 
 scap_userinfo *sinsp_usergroup_manager::add_user(const std::string &container_id, int64_t pid, uint32_t uid, uint32_t gid, std::string_view name, std::string_view home, std::string_view shell, bool notify)
 {
+	// ignore NSS entries
```

Comment:
I am still not convinced this is the right place: add_user() is called with empty params by `sinsp_threadinfo::set_user`; it is called with full params only when reading from scap files (`sinsp_parser::parse_user_evt`).
I think we should actually chekc inside `add_host_user` and `add_container_user` when we call `__getpwuid` and `fgetpwent`.
Same for groups.

Reply:
@fededp I'm ready to move the implementation if there's a better way, however here's my rationale and if I understand correctly it would be too late inside the functions you propose.
 
I've put the check there because `add_host_user` and `add_container_user` come after this check:
```c++
	scap_userinfo *usr = get_user(container_id, uid);
	if(usr)
	{
		// Update user if it was already there
		if (name.data() != nullptr)
		{
			strlcpy(usr->name, std::string(name).c_str(), MAX_CREDENTIALS_STR_LEN);
			strlcpy(usr->homedir, std::string(home).c_str(), SCAP_MAX_PATH_SIZE);
			strlcpy(usr->shell, std::string(shell).c_str(), SCAP_MAX_PATH_SIZE);
		}
		return usr;
	}
```
In the case of NSS users, this is actually finding the _root_ user as the _uid_ is zero, thus returning early - before `add_*_user` - overwriting the details of _root_.

> add_user() is called with empty params by `sinsp_threadinfo::set_user`

In this case we should be fine as the name would be empty, hence not starting with a `+` or a `-` and the new code would be a no-op.

Reply:
I understand now, that works fine, you are right!